SandWorm/SandWormComponent.cs

```diff
@@ -79,6 +82,34 @@ protected override void RegisterOutputParams(GH_Component.GH_OutputParamManager
             pManager.AddTextParameter("Output", "O", "Output", GH_ParamAccess.list); //debugging
         }
 
+        protected override void AppendAdditionalComponentMenuItems(ToolStripDropDown menu)
+        {
+            base.AppendAdditionalComponentMenuItems(menu);
+            Menu_AppendItem(menu, "Color Mesh by Elevation", SetMeshColorStyle, true, selectedColorStyle == MeshColorStyle.byElevation);
+            menu.Items[menu.Items.Count - 1].Tag = MeshColorStyle.byElevation;
+            Menu_AppendItem(menu, "Do Not Color Mesh", SetMeshColorStyle, true, selectedColorStyle == MeshColorStyle.noColor);
+            menu.Items[menu.Items.Count - 1].Tag = MeshColorStyle.noColor;
+        }
+
+        private void SetMeshColorStyle(object sender, EventArgs e)
+        {
+            ToolStripMenuItem selectedItem = (ToolStripMenuItem)sender;
+            ToolStrip parentMenu = selectedItem.Owner as ToolStrip;
+            if ((MeshColorStyle)selectedItem.Tag != selectedColorStyle) // Update selected style if it has changed
+            {
+                selectedColorStyle = (MeshColorStyle)selectedItem.Tag;
+                ExpireSolution(true);
```

Comment:
Clearing vertexColors here should fix the ghosting issue. Please test, though.
`quadMesh.VertexColors.Clear();`